airflow/jobs/scheduler_job.py

```diff
@@ -1408,6 +1416,40 @@ def _send_sla_callbacks_to_processor(self, dag: DAG) -> None:
         )
         self.executor.send_callback(request)
 
+    @provide_session
+    def _fail_tasks_stuck_in_queued(self, session: Session = NEW_SESSION) -> None:
+        """
+        Mark tasks stuck in queued for longer than `task_queued_timeout` as failed.
+
+        Tasks can get stuck in queued for a wide variety of reasons (e.g. celery loses
+        track of a task, a cluster can't further scale up its workers, etc.), but tasks
+        should not be stuck in queued for a long time. This will mark tasks stuck in
+        queued for longer than `self._task_queued_timeout` as failed. If the task has
+        available retries, it will be retried.
```

Comment:
I don't see any handling of retry state in this code, we go straight to failed and bypass retry logic.

Reply:
Would it be sufficient to call `TI.handle_failure()`?

Reply:
It seems like it should be based on my read of `TI.handle_failure`.